blocks/components/editable/index.js

```diff
@@ -73,6 +77,32 @@ export default class Editable extends wp.element.Component {
 		this.props.onChange( value );
 	}
 
+	isStartOfEditor() {
+		const range = this.editor.selection.getRng();
+		if ( range.startOffset !== 0 || ! range.collapsed ) {
+			return false;
+		}
+		const start = range.startContainer;
+		const body = this.editor.getBody();
+		let element = start;
+		do {
+			const child = element;
+			element = element.parentNode;
+			if ( element.childNodes[ 0 ] !== child ) {
+				return false;
+			}
+		} while ( element !== body );
+		return true;
+	}
+
+	onKeyDown( event ) {
+		if ( this.props.onMerge && event.keyCode === KEYCODE_BACKSPACE && this.isStartOfEditor() ) {
```

Comment:
I don't know if it was intentional, but nice optimization having the heaviest part of the condition performed last and taking advantage of short-circuiting 👍